lib/incoming_mail.rb

```diff
@@ -0,0 +1,110 @@
+module IncomingMail
+  # mail_types
+  UNKNOWN = 0
+  REQUEST = 1
+
+  class Handler
+    # parses a Mandrill callback object and Does The Right Thing
+
+    def initialize(params = {})
+      params.each do |key, value|
+        instance_variable_set("@#{key}", value)
+      end
+    end
+
+    def handle(payload)
+      if payload.is_a?(Array) and payload[0]['event'] and payload[0]['event'] == 'inbound'
+        create_response(payload[0])
+      elsif payload.is_a?(Mandrill::WebHook::EventDecorator)
+        create_response(payload)
+      else
+        fail "Invalid Mandrill event payload. Must be event==inbound"
+      end
+    end
+
+    private
```

Comment:
I believe `private` has no effect in modules -- another reason to re-CLASSify this file ;)